stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -1490,14 +1490,10 @@ end
     n == 0 && return zero(eltype(x))
     @inbounds begin
         ξ1 = x[1]
-        normu = abs2(ξ1)
-        for i = 2:n
-            normu += abs2(x[i])
-        end
+        normu = norm(x)
```

Comment:
Note that `norm` is actually probably a bit slower here because it has additional checks for overflow.   Of course, that may be a good thing!

Reply:
You're right. I'd say it's more in line with dimension 2 where we have Given rotations computed through a julia translation of LAPACKs lartg, which is an expensive way to compute a stable reflector.